### DISCUSSION TITLE
MOOSE/Moltres memory usage

### DISCUSSION THREAD

**Discussion by sychang2021** [arfc/moltres#209] [Q&A Software development] [answered]
Hi, everyone.

I want to calculate a case with a lot of variables. It has 46,800 elements, 17 variables and 570,000 degrees of freedom. 
![6074b70d3d0552399a04fb318a89be2](https://user-images.githubusercontent.com/104487015/177839874-ce3f3c8b-5100-4bab-8678-c363e16b4834.png)
I used the `lu` method and the Executioner settings are as follows:
```
  type = Transient
  num_steps = 100
  solve_type = 'NEWTON'
  dt = 2e-7
  #end_time = 100

  nl_rel_tol = 1e-6
  nl_abs_tol = 1e-6

  solve_type = 'NEWTON'
  petsc_options = '-snes_converged_reason -ksp_converged_reason -snes_linesearch_monitor'
  petsc_options_iname = '-pc_type -pc_factor_shift_type'
  petsc_options_value = 'lu       NONZERO'
  line_search = 'none'
   # petsc_options_iname = '-snes_type'
  # petsc_options_value = 'test'

  nl_max_its = 30
  l_max_its = 100
  #steady_state_detection = true
  #steady_state_tolerance = 1e-10

  automatic_scaling = true
```
In fact this is a small single rod model and the number of elements is very small.  Even so， when I run it on a computer with 64GB of RAM, the RAM is full and no results are produced. If I want to use MOOSE to calculate relatively large problems, it is obviously not feasible, such as the problem of six groups of neutrons and six groups of recursors with millions of grids.

I have tried `ilu` but the results do not converge, and only when using `lu` (maybe set to other options will also converge). Can you give me some advice on saving memory and guaranteeing convergence?

**Comment by sychang2021** [comment]
There is another question about `scaling` . I want to use `automatic_scaling = true` in `Executioner` block. At the beginning of the neutron calculation, if the neutron flux and the precursors have small values, their residuals are very small. Will this cause MOOSE to think that the iteration has converged, which will affect the accuracy of the solution?

**Comment by smpark7** [comment, marked as answer]
64GB RAM shared by 8 processors should be more than enough for a problem with 570,000 degrees of freedom. You may refer to this [page](https://arfc.github.io/software/moltres/wiki/recommended_settings/) for recommendations on the petsc options. `asm` with `lu` as the subsolver will require less memory. `ilu` is usually not good enough to resolve the strong coupling between the flux and precursor variables.

`automatic_scaling` is good for the problem you're describing. If the flux and precursor values are small, `automatic_scaling` scales their residual or jacobian values up to approximately unity (~1). You'll also want `compute_scaling_once = false` so that Moltres adjusts the scaling factor at every timestep.
